src/charm.py

```diff
@@ -738,10 +843,108 @@ def _get_initialization_secret_from_peer_relation(self) -> Tuple[str, List[str]]
         except (TypeError, SecretNotFoundError, AttributeError):
             raise PeerSecretError(secret_name=VAULT_INITIALIZATION_SECRET_LABEL)
 
+    def _create_s3_session(self, s3_parameters: dict) -> Optional[boto3.session.Session]:
```

Comment:
I suspect the answer is yes since we're using it but `boto3` should work for non-aws S3?

Reply:
Yes it works, tested on `Minio`.